### ISSUE TITLE
Offline Problems

### ISSUE THREAD

**Issue by NyanDynamiX** [c0dycode/BL-AutoexecDLL#10]
Online works flawlessly (it auto patches and all that) however if im not connected to the internet and steam is in offline mode, it seems to not detect the offline version of the patches. I tried renaming the offline patch and checking the autoexec.txt but it doesn't seem to auto exec the offline version of UCP 4.2 (online is fine). I tried removing and making 'forceofflinefixes' true and false but it still doesn't work regardless of configuration.

Any ideas on what could be the problem and how i could fix it? (I'm currently at my relatives who lives far in the mountain regions that has crappy net so I'm usually playing steam offline and i kinda forget sometimes to patch manually since im so used to the autoexec working xd)

Note that previous versions prior to the new DLCs worked online and offline.

My autoexec:

Online=exec Patch.txt
Offline=exec PatchOffline.txt
Fastmode=true

EDIT: To avoid confusion in my wording, **the autoexec works offline, its just not executing the offline patch.** manually inputting it works though.

**Comment by c0dycode** [comment]
It seems that the place I check whether the player is in the main menu or not, is not being used when steam is in offline mode. I can't remember if it worked before. Could've been an oversight. Try this version :)

I think this fixes it :)
[autoexec-UHD.zip](https://github.com/c0dycode/BL-AutoexecDLL/files/3298262/autoexec-UHD.zip)

**Comment by NyanDynamiX** [comment]
I tried that and I'm afraid to say it still wont load the offline patch automatically.  Manually inputting still works as intended. Could it be my autoexec.txt? I added the 'ForceOfflineHotFixes' line underneath 'Fastmode' and set the former to false.

**Comment by c0dycode** [comment]
Yes, if ForceOfflineHotfixes is not set to true, then it'll automatically detect, whether you can use the Online patch. If you can not, then it'll execute the Offline patch. If you force Offline, then it should automatically execute the Offline patch.

**Comment by NyanDynamiX** [comment]
I'll try removing the line again and test it.

**Comment by NyanDynamiX** [comment]
I tried removing and setting 'Forceofflinehotfixes' to false and true, still won't execute the patch. Manually inputting still works as intended. Could it be something with fastmode?

**Comment by c0dycode** [comment]
This is with the version attached above. For me it works just fine as you can see 🤔 
https://gfycat.com/UnfoldedInformalFlycatcher

Make sure you double check the spelling. Capital letters also matter for "Online=, Offline=, Fastmode= and ForceOfflineHotfixes="

**Comment by NyanDynamiX** [comment]
Few things i've noticed:

1) You are correct, setting it on True and checking the spelling and all will make it work offline
2) Fast mode sometimes has a rare chance of not auto patching (most likely related to my system being a bit potato-ish)
3) It will only work if offlinehotfixes is set to true, if i set it to false it wont detect if I'm online or not and wont auto exec the offline patch. Prior versions didnt have this issue from my experience since i didn't include the offlinehotfixes line in my autoexec.txt on older versions (before Commander Lil DLC) 

Will I have to change it to false if I play online?

**Comment by c0dycode** [comment]
Oh, looks like not forcing it fails to properly detect it now. That's odd. I'll check that out. In the meantime, you can force Offline just fine, even if you play online. It'll basically just block gearbox from applying their default hotfixes. Those are already included by default in UCP, Reborn etc.

**Comment by NyanDynamiX** [comment]
Oh so UCP has gearbox's fixes already? I did not know that... Is there a difference though? I'm kinda curious. xD

**Comment by NyanDynamiX** [comment]
Anyways thanks for the help! Good luck on finding the problem!

**Comment by c0dycode** [comment]
Yes, pretty much any mod that has been made via BLCMM will contain the Gearbox Hotfixes. :)
BLCMM should automatically include them.
